### PR TITLE
CI: clean install & share pkg build

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

- share building stage
- test package at completely clean env

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @carmocca @akihironitta

### CODE REVIEW

**Comment by Borda** [comment]
waiting for #15984

**Comment by Borda** [comment]
it tries to load UI from the local destination instead of the one from the installed package:
```
FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/lightning/lightning/src/lightning_app/ui/index.html'
```

**Review by awaelchli — APPROVED** [review]

File: .github/workflows/ci-pkg-install.yml [hunk 1]
```diff
@@ -28,52 +31,71 @@ defaults:
...
+          python .actions/assistant.py copy_replace_imports --source_dir="./src" \
+            --source_import="pytorch_lightning,lightning_lite,lightning_app" \
+            --target_import="lightning.pytorch,lightning.lite,lightning.app"
+          rm -rf src/lightning
+      - name: Rename src folders
+        working-directory: src/
+        run: |
+          mv pytorch_lightning pl
+          mv lightning_lite lit_lite
+          mv lightning_app lit_app
```
Comment (carmocca) [inline .github/workflows/ci-pkg-install.yml:86]: Can you explain why move these directories?

File: .github/workflows/ci-pkg-install.yml [hunk 2]
```diff
@@ -28,52 +31,71 @@ defaults:
...
         run: lightning --version
 
+      - name: Adjust code for Lit
+        if: ${{ matrix.pkg-name == 'lightning' || matrix.pkg-name == 'notset' }}
+        run: |
+          pip install -q -r .actions/requirements.txt
+          python .actions/assistant.py copy_replace_imports --source_dir="./src" \
+            --source_import="pytorch_lightning,lightning_lite,lightning_app" \
+            --target_import="lightning.pytorch,lightning.lite,lightning.app"
+          rm -rf src/lightning
```
Comment (carmocca) [inline .github/workflows/ci-pkg-install.yml:80]: Why remove the source code?

File: .github/workflows/ci-pkg-install.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/ci-pkg-install.yml:86]: so there is no accidental use of local version but importing the install version

File: .github/workflows/ci-pkg-install.yml (hunk 2 above)
Comment (Borda) [reply .github/workflows/ci-pkg-install.yml:80]: because I want to test the main package on all subset of our doctests

Comment (Borda) [reply .github/workflows/ci-pkg-install.yml:80]: see for example: https://github.com/Lightning-AI/lightning/actions/runs/3681710895/jobs/6228712319

**Pull request merged**